cmake/projects/flex/schemes/url_sha1_flex_autotools.cmake.in

```diff
@@ -0,0 +1,51 @@
+cmake_minimum_required(VERSION 3.0)
+project(Hunter)
+
+list(APPEND CMAKE_MODULE_PATH "@HUNTER_SELF@/cmake/modules")
+
+include(hunter_autotools_project)
+include(hunter_status_debug)
+include(hunter_test_string_not_empty)
+include(hunter_user_error)
+
+hunter_status_debug("Scheme: url_sha1_flex_autotools")
+
+if("@MSVC@")
+  hunter_user_error("Autotools scheme not supported with Visual Studio")
+endif()
+
+# Check preconditions
+hunter_test_string_not_empty("@HUNTER_SELF@")
+hunter_test_string_not_empty("@HUNTER_EP_NAME@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_URL@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_SHA1@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_DOWNLOAD_DIR@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_SOURCE_DIR@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_BUILD_DIR@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_INSTALL_PREFIX@")
+hunter_test_string_not_empty("@HUNTER_INSTALL_PREFIX@")
+
+hunter_autotools_project("@HUNTER_EP_NAME@"
+    HUNTER_SELF
+      "@HUNTER_SELF@"
+    URL
+      @HUNTER_PACKAGE_URL@
+    URL_HASH
+      SHA1=@HUNTER_PACKAGE_SHA1@
+    DOWNLOAD_DIR
+      "@HUNTER_PACKAGE_DOWNLOAD_DIR@"
+    SOURCE_DIR
+      "@HUNTER_PACKAGE_SOURCE_DIR@"
+    BUILD_DIR
+      "@HUNTER_PACKAGE_BUILD_DIR@"
+    INSTALL_DIR
+      "@HUNTER_PACKAGE_INSTALL_PREFIX@"
+    GLOBAL_INSTALL_DIR
+      "@HUNTER_INSTALL_PREFIX@"
+    PARALLEL_JOBS
+      "@HUNTER_JOBS_OPTION@"
+    PACKAGE_CONFIGURATION_TYPES
+      "@HUNTER_PACKAGE_CONFIGURATION_TYPES@"
+    CPPFLAGS
+      "-D_GNU_SOURCE"
```

Comment:
This line is the only reason I have used a custom scheme.

Reply:
Looks like something that should be defined in sources:
```
You should define these macros by using ‘#define’ preprocessor directives at
the top of your source code files. These directives must come before any
#include of a system header file. It is best to make them the very first thing
in the file, preceded only by comments. You could also use the ‘-D’ option to
GCC, but it’s better if you make the source files indicate their own meaning in
a self-contained way.
```


From documentation:
* https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html

Reply:
I would do that but it is a generated file that has the main issue. They solve the problem in a number of upstream commits but that won't be released until they finalize version 2.6.5. I was hoping to avoid cherry-picking the specific commits.